types/frida-gum/index.d.ts

```diff
@@ -0,0 +1,6668 @@
+     * garbage-collected or the script is unloaded.
+     */
+    close(): void;
+
+    /**
+     * Executes a raw SQL query. Throws an exception if the query is invalid.
+     *
+     * The query's result is ignored, so this should only be used for queries
+     * for setting up the database, e.g. table creation.
+     *
+     * @param sql Text-representation of the SQL query.
+     */
+    exec(sql: string): void;
+
+    /**
+     * Compiles the provided SQL into a `SqliteStatement` object. Throws an
+     * exception if the query is invalid.
+     *
+     * @param sql Text-representation of the SQL query.
+     */
+    prepare(sql: string): SqliteStatement;
+
+    /**
+     * Dumps the database to a gzip-compressed blob encoded as Base64.
+     *
+     * This is useful for inlining a cache in your agent's code, loaded by
+     * calling `SqliteDatabase.openInline()`.
+     */
+    dump(): string;
+}
+
+declare interface SqliteOpenOptions {
+    flags?: SqliteOpenFlag[];
+}
+
+declare const enum SqliteOpenFlag {
+    ReadOnly = "readonly",
+    ReadWrite = "readwrite",
+    Create = "create",
+}
+
+/**
+ * Pre-compiled SQL statement.
+ */
+declare class SqliteStatement {
+    /**
+     * Binds the integer `value` to `index`.
+     *
+     * @param index 1-based index.
+     * @param value Integer value to bind.
+     */
+    bindInteger(index: number, value: number): void;
+
+    /**
+     * Binds the floating point `value` to `index`.
+     *
+     * @param index 1-based index.
+     * @param value Floating point value to bind.
+     */
+    bindFloat(index: number, value: number): void;
+
+    /**
+     * Binds the text `value` to `index`.
+     * @param index 1-based index.
+     * @param value Text value to bind.
+     */
+    bindText(index: number, value: string): void;
+
+    /**
+     * Binds the blob `bytes` to `index`.
+     *
+     * @param index 1-based index.
+     * @param bytes Blob value to bind.
+     */
+    bindBlob(index: number, bytes: ArrayBuffer | number[] | string): void;
+
+    /**
+     * Binds a `null` value to `index`.
+     *
+     * @param index 1-based index.
+     */
+    bindNull(index: number): void;
+
+    /**
+     * Either starts a new query and gets the first result, or moves to the
+     * next one.
+     *
+     * Returns an array containing the values in the order specified by the
+     * query, or `null` when the last result is reached. You should call
+     * `reset()` at that point if you intend to use this object again.
+     */
+    step(): any[] | null;
+
+    /**
+     * Resets internal state to allow subsequent queries.
+     */
+    reset(): void;
+}
+
+/**
+ * Intercepts execution through inline hooking.
+ */
+declare namespace Interceptor {
+    /**
+     * Intercepts calls to function/instruction at `target`. It is important
+     * to specify a `InstructionProbeCallback` if `target` is not the first
+     * instruction of a function.
+     */
+    function attach(target
```

Comment:
This could also be written as a union type: `type Arm64ConditionCode = "eq" | "ne" | ...`

Reply:
Yes I was tempted to do that, but leaned towards using an enum to make the code clearer.